chalice/config.py

```diff
@@ -76,22 +110,18 @@ def chalice_app(self):
         return self._chain_lookup('chalice_app')
 
     @property
-    def autogen_policy(self):
-        # type: () -> bool
-        return self._chain_lookup('autogen_policy')
-
-    @property
     def config_from_disk(self):
         # type: () -> StrMap
         return self._config_from_disk
 
-    def _chain_lookup(self, name):
-        # type: (str) -> Any
-        all_dicts = [
-            self._user_provided_params,
-            self._config_from_disk,
-            self._default_params
-        ]
+    def _chain_lookup(self, name, varies_per_chalice_stage=False):
+        # type: (str, bool) -> Any
+        all_dicts = [self._user_provided_params]
```

Comment:
all_dicts is probably not the best name considering that another dict is added to it if `varies_per_chalice_stage` is true. How bout something like search_dicts?